src/mimir_coordinator.py

```diff
@@ -37,37 +42,39 @@
     MimirRole.compactor: 1,
     MimirRole.distributor: 1,
 }
-"""The set of roles that need to be allocated for the 
+"""The set of roles that need to be allocated for the
 deployment to be considered robust according to the official recommendations/guidelines."""
 
 
 class MimirCoordinator:
+    """Mimir coordinator."""
+
     def __init__(self, relations: List[Relation]):
         self.relations = relations
 
     def is_coherent(self):
         """Return True if the roles list makes up a coherent mimir deployment."""
         roles = self.roles()
-        return set(roles).issubset(MINIMAL_DEPLOYMENT)
+        return roles and set(roles).issubset(MINIMAL_DEPLOYMENT)
 
-    def is_recommended(self):
+    def is_recommended(self) -> bool:
+        """Return True if is a subset of the minimal deployment."""
         roles = self.roles()
-        return set(roles).issubset(MINIMAL_DEPLOYMENT)
+        return bool(roles) and set(roles).issubset(MINIMAL_DEPLOYMENT)
 
-    def roles(self) -> Counter[str]:
+    def roles(self) -> Counter:
         """Gather the roles from the mimir_worker relations and count them."""
         roles = Counter()
 
         for relation in self.relations:
             if not self._relation_data_valid(relation):
-                logger.error(f'Invalid relation data in {relation!r}')
+                logger.error(f"Invalid relation data in {relation!r}")
```

Comment:
Not so critical, but:

```suggestion
                logger.error("Invalid relation data in %s", relation)
``